rclcpp/include/rclcpp/exceptions.hpp

```diff
@@ -111,6 +111,7 @@ class InvalidServiceNameError : public NameValidationError
  * \throws RCLErrorBase some child class exception based on ret
  */
 RCLCPP_PUBLIC
+[[noreturn]]
```

Comment:
Clang and Visual studio seem to be unhappy with `[[noreturn]]` here

* https://ci.ros2.org/job/ci_osx/5651/console#console-section-7
* https://ci.ros2.org/job/ci_windows/6702/console#console-section-6

Could it be `RCLCPP_PUBLIC` has some content on these platforms and the compiler thinks that's what the attribute list applies to? Maybe they would be happy with

```c++
RCLCPP_PUBLIC
void
throw_from_rcl_error [[noreturn]] (...`
```

Reply:
That's a good idea, at least one of the cppreference examples do that:

https://en.cppreference.com/w/cpp/language/attributes/noreturn

Reply:
It worked on Windows. Thanks!